server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -353,40 +351,39 @@ private boolean mergeParam(RuleParamDto paramDto, RulesDefinition.Param paramDef
     return changed;
   }
 
-  private static boolean mergeTags(RulesDefinition.Rule ruleDef, RuleDto dto) {
+  private static boolean mergeTags(RulesDefinition.Rule ruleDef, RuleDefinitionDto dto) {
     boolean changed = false;
 
     if (RuleStatus.REMOVED == ruleDef.status()) {
-      dto.setSystemTags(Collections.<String>emptySet());
+      dto.setSystemTags(Collections.emptySet());
       changed = true;
     } else if (dto.getSystemTags().size() != ruleDef.tags().size() ||
-      !dto.getSystemTags().containsAll(ruleDef.tags())
-      || !intersection(dto.getTags(), ruleDef.tags()).isEmpty()) {
+      !dto.getSystemTags().containsAll(ruleDef.tags())) {
       dto.setSystemTags(ruleDef.tags());
-      // remove end-user tags that are now declared as system
-      RuleTagHelper.applyTags(dto, ImmutableSet.copyOf(dto.getTags()));
+      // FIXME this can't be implemented easily with organization support: remove end-user tags that are now declared as system
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)